### PR TITLE
Fix parameter order and bump to 1.0.0.

### PR DESCRIPTION
**Pull request by patrickt**
Also adds a `queryOnly` convenience function.

### CODE REVIEW

File: postgresql-transactional.cabal [hunk 1]
```diff
@@ -17,7 +23,7 @@ source-repository head
 library
   exposed-modules:     Database.PostgreSQL.TransactionalStore
   hs-source-dirs: src
-  ghc-options: -Wall -Werror
+  ghc-options: -Wall
```
Comment (patrickt) [inline postgresql-transactional.cabal]: Just a heads up: `stack sdist` whines at you if you have `-Werror` on, with the justification that new versions of GHC can produce new warnings that will make builds fail.

File: src/Database/PostgreSQL/Transaction.hs [hunk 2]
```diff
@@ -115,14 +121,19 @@ executeOne :: (ToRow input, MonadIO m)
...
-  results <- execute q params
-  return (results == 1)
+executeOne q params = (== 1) <$> execute q params
+
+-- | Lookup a single FromField value. This takes care of handling 'Only' for you.
+queryOnly :: (ToRow params, FromField f, MonadIO m)
+          => params
+          -> Postgres.Query
+          -> PGTransactionT m (Maybe f)
+queryOnly q params = fmap Postgres.fromOnly <$> queryHead q params
```
Comment (lseppala) [inline src/Database/PostgreSQL/Transaction.hs]: Swap params

File: src/Database/PostgreSQL/Transaction.hs [hunk 3]
```diff
@@ -82,29 +88,29 @@ query_ q = ask >>= liftIO . (`Postgres.query_` q)
...
-returning q params = ask >>= (\conn -> liftIO $ Postgres.returning conn q params)
+returning params q = ask >>= (\conn -> liftIO $ Postgres.returning conn q params)
 
 -- | Run a query and return 'Just' the first result found or 'Nothing'.
 queryHead :: (ToRow input, FromRow output, MonadIO m)
           => input
           -> Postgres.Query
           -> PGTransactionT m (Maybe output)
-queryHead params q = do
+queryHead q params = do
```
Comment (lseppala) [inline src/Database/PostgreSQL/Transaction.hs]: swap params

**Comment by lseppala** [comment]
👍

File: src/Database/PostgreSQL/Transaction.hs [hunk 4]
```diff
@@ -115,14 +121,19 @@ executeOne :: (ToRow input, MonadIO m)
...
+queryOnly :: (ToRow params, FromField f, MonadIO m)
+          => params
+          -> Postgres.Query
+          -> PGTransactionT m (Maybe f)
+queryOnly params q = fmap Postgres.fromOnly <$> queryHead params q
 
 formatQuery :: (ToRow q, MonadIO m)
-            => Postgres.Query
-            -> q
+            => q
```
Comment (lseppala) [inline src/Database/PostgreSQL/Transaction.hs]: Kinda confusing that `q` is the type parameter but `query` is the bound parameter

Comment (lseppala) [inline src/Database/PostgreSQL/Transaction.hs]: we used `input` previously

**Comment by lseppala** [comment]
:+1:

**Pull request merged**